py/torch_tensorrt/dynamo/conversion/impl/dynamic_block_quantize.py

```diff
@@ -0,0 +1,250 @@
+                ctx,
+                target,
+                source_ir,
+                name,
+                input_tensor,
+                global_scale,
+                axis,
+            )
+
+        else:
+            raise ValueError(
+                f"quantizer received an input of {name}. Supported values: weight_quantizer | input_quantizer"
+            )
+        return output
+
+
+def _dynamic_double_quantize(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input_tensor: TRTTensor,
+    global_scale: torch.Tensor,
+    axis: int = -1,
+    block_size: int = 16,
+    output_type: trt.DataType = trt.DataType.FP4,
+    scale_type: trt.DataType = trt.DataType.FP8,
+) -> TRTTensor:
+    """
+    quantize input tensor to fp4
+    Parameters:
+        ctx: ConversionContext,
+        target: Target,
+        source_ir: Optional[SourceIR]
+        name: str
+        input_tensor : TRTTensor (On GPU)
+            The input TRTTensor.
+        global_scale : Tensor (On GPU)
+            The global per-tensor scaling factor. It should contain only 1 element.
+        axis : int
+            The axis to quantize. Default is -1 (the last axis).
+        block_size : int
+            The block size for quantization. Default is 16.
+        output_type : trt.DataType
+            The data type for quantized data. Default is FP4.
+        scale_type : trt.DataType
+            The data type for block scale. Default is FP8.
+
+    """
+    global_scale = get_trt_tensor(ctx, global_scale, name + "_global_scale")
+
+    if input_tensor.dtype not in [
+        trt.DataType.HALF,
+        trt.DataType.FLOAT,
+        trt.DataType.BF16,
+    ]:
+        raise ValueError(
+            f"Currently supported input tensor type is float16 | float32 | bfloat16, got Unsupported dtype: {input_tensor.dtype}"
+        )
+    # dynamic quantize input tensor to fp4
+    dynamic_quantize_layer = ctx.net.add_dynamic_quantize(
+        input_tensor,
+        axis,
+        block_size,
+        output_type,
+        scale_type,
+    )
+    dynamic_quantize_layer.set_input(1, global_scale)
+    set_layer_name(
+        dynamic_quantize_layer, target, name + "_dynamic_quantize", source_ir
+    )
+    quantized_data_in_fp4 = dynamic_quantize_layer.get_output(0)
+    quantized_scale_in_fp8 = dynamic_quantize_layer.get_output(1)
+
+    return _double_dequantize(
+        ctx,
+        target,
+        source_ir,
+        name,
+        quantized_data_in_fp4,
+        quantized_scale_in_fp8,
+        global_scale,
+        axis,
+        input_tensor.dtype,
+    )
+
+
+def _double_dequantize(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    quantized_data_in_fp4: TRTTensor,
+    quantized_scale_in_fp8: TRTTensor,
+    global_scale: torch.Tensor,
+    axis: int = -1,
+    output_type: trt.DataType = trt.DataType.FLOAT,
```

Comment:
minor : add a docstring and a little bit of explanation about the function wherever possible. Any references to this quantization (eg: TRT docs) will be useful